Initial Draft for Issue 269

NaraHita opened the pull request:
Fixes #269 

Added new Document to provide an Overview for ODH-015 - Docs as Code.

NaraHita committed:
Initial Draft for Issue 269

NaraHita committed:
Add table of contents to ODH-015 Docs as Code

Added a table of contents to the 'Docs as Code' document and made minor formatting adjustments.

NaraHita committed:
Remove Table of Content from ODH-015 documentation

Removed the Table of Content section from the documentation.

NaraHita committed:
Add 'Documentation as Code' topic to discussions

NaraHita committed:
Merge branch 'canonical:main' into 269_Doc_as_Code

Sophie-Pages commented on website/events/community-hour-discussions/odh-015-docs-as-code.md:
```diff
@@ -0,0 +1,82 @@
+# Overview of Documentation Community Hour[ODH-015]: Documentation as Code
```
Suggestion and example:
For consistency, the article should follow how we usually format the file name, title, or the beginning of the document. I'll let you take a look at the rest of the document to see if anything else needs to be changed for consistency.

NaraHita replied:
Ok, will update this.

Sophie-Pages commented on website/events/community-hour-discussions/odh-015-docs-as-code.md:
```diff
@@ -0,0 +1,82 @@
+
+## What is *Docs as Code*?
+
+*Docs as Code* is an approach that treats documentation as a first-class citizen alongside software source code. Instead of favoring code, it places documentation on equal footing in the software development lifecycle.
```
Question:
- What do you mean by "treats documentation as a first-class citizen"? Why have you decided to use that as the definition?
- Also, I don't think Docs as Code is about placing "documentation on equal footing in the software development lifecycle". I can happen, but I don't think it's one of the main reasons we use this approach. What do you think?

NaraHita replied:
Hi @Sophie-Pages , so in the video Robert mentions around the 5 minute mark that Docs as Code means treating Documentation same as software code. I just picked up on that point and try to emphasis with a simile. But, I can simplify this definition to make it more accessible. That is cool!

Sophie-Pages commented on website/events/community-hour-discussions/odh-015-docs-as-code.md:
```diff
@@ -0,0 +1,82 @@
+
+## Introduction
+
+Many of us in tech have experienced the challenge of writing both code and documentation, often favoring code over documentation. The philosophy of Documentation as Code, more commonly known as _**Docs as Code**_, bridges this gap by treating documentation with the similar tools, processes, and respect as source code. In this session, Robert Krátký explains how this approach streamlines documenta ...
```
Suggestion:
When you say it "bridges this gap", do you mean that it bridges the gap between writing code and documentation?
I would remove the "often favoring code over documentation", to make the introduction more concise and also because it might not be the case for some readers. What do you think?

```suggestion
The philosophy of Documentation as Code, more commonly known as _**Docs as Code**_, bridges the gap between software development and documentation by treating documentation with software tools and processes. In this session, Robert Krátký explains how this approach streamlines documentation development by integrating it closely with source code in software projects.
```

Sophie-Pages commented on website/events/community-hour-discussions/odh-015-docs-as-code.md:
```diff
@@ -0,0 +1,82 @@
+| Version Control      | Git, SVN, Mercurial                       | Essential for tracking changes and history       					|
+| Issue Tracking       | JIRA, Bugzilla, Mantis                    | Manage bugs and improvements                      |
+| Testing & Validation | Scripts, Linters, Spell Checkers          | Borrow software testing concepts to ensure quality and reusability |
+| Publishing           | Static Site Generators, CMS               | Automated hosting and updates                      |
```
Suggestions
- Suggestion to clean the table (hopefully it should still work!).
- I haven't made the modifications, but what do you think of reviewing the case of some words? For example, I don't think "Issue Tracking", "Spell Checkers", or "Text Editors" need to be in title case. 
- Also, I don't think all our users know what CMS means. When I first use an acronym, I recommend that we spell it out and use the acronym in parentheses. The acronym can then be used throughout the document. Could you please review the document to define acronyms? Thank you!

```suggestion
| Goals | Tools | Reasoning |
|---|---|---|
| Content Authoring | Text Editors | Use non-binary, human-readable formats |
| Formatting & Styling | Markup Languages | Semantic formatting with stylesheets enables multiple output formats (HTML, PDF) |
| Version Control | Git, SVN, Mercurial | Essential for tracking changes and history |
| Issue Tracking | JIRA, Bugzilla, Mantis | Manage bugs and improvements |
| Testing & Validation | Scripts, Linters, Spell Checkers | Borrow software testing concepts to ensure quality and reusability |
| Publishing | Static Site Generators, CMS | Automated hosting and updates |
```

Sophie-Pages commented on website/events/community-hour-discussions/odh-015-docs-as-code.md:
```diff
@@ -0,0 +1,82 @@
+  
+- Developers already know these tools- including IDEs, text editors, and scripting tools- making it more natural for them to contribute to documentation, fostering stronger collaboration.
+
+### The Bad and the Ugly
```
Question and suggestion:
- Why have you decided to use "The good" and "The bad and the ugly" as headers?
- I recommend to put headers in sentence case (only the first letter is capitalized) if they don't relate to acronyms (Content Management Systems), product names (Ubuntu Pro), program names, etc. You can find more about this in the Canonical documentation style guide (https://docs.ubuntu.com/styleguide/en/#headings).

NaraHita replied:
Was just trying to make it an interesting read, by using a famous American movie reference. 
Will remove it and make it positives and negatives.

Sophie-Pages replied:
I like the idea of adding your personality inside the document! For that specific case, some users may not have the reference and may be taken aback by "The bad and the ugly". I think positives/negatives or benefits/challenges may be more understandable.

Sophie-Pages commented on website/events/community-hour-discussions/odh-015-docs-as-code.md:
```diff
@@ -0,0 +1,82 @@
+1. Efforts are underway to streamline collaboration with external projects.
+2. Automation is being developed to verify Contributor License Agreement(CLA) before contributors can submit pull requests to the Open Documentation Repository.
+3. The Open Documentation schedule is available on the Forum to plan future Office Hour discussion topics.
+4. Next week’s session will feature Nick Veitch discussing *Vale* for testing documentation.
```
Question:

This part doesn't relate to Documentation as Code. Why have you decided to include it?

NaraHita replied:
Hi @Sophie-Pages, I can remove this part if needed. I thought it was part of the initial Issue that any other callouts must be included too. Please comment if you want me to remove it.

Sophie-Pages replied:
I think it would be better to keep our article focused on the initial topic and to remove this. My initial issue might not have been clear enough, sorry about that. I think it's great to add other callouts, but only if they relate to the topic.

NaraHita replied:
Sure. Will do that.

Sophie-Pages commented on website/events/community-hour-discussions/odh-015-docs-as-code.md:
```diff
@@ -0,0 +1,82 @@
+
+*Docs as Code* is an approach that treats documentation as a first-class citizen alongside software source code. Instead of favoring code, it places documentation on equal footing in the software development lifecycle.
+
+This approach avoids reliance on a single monolithic system, it leverages specialized tools throughout the content creation, review, and publishing process. Often, the choice of tools aligns with the specific software process it addresses.
```
Suggestion:

I would suggest to either define what a "monolithic system" is or to redo the sentence without it to make it more accessible. Some users might not understand what a "single monolithic system" is. What do you think?

NaraHita replied:
Again, the "monolithic system" reference comes from the video. But, I will add a definition for accessibility.

Sophie-Pages replied:
Yes, you're absolutely right: it was shared without additional information in the video. In that case and to ensure our users can understand what we are explaining, I think it's fine to extend a bit Robert's video and ideas.

Sophie-Pages commented on website/events/community-hour-discussions/odh-015-docs-as-code.md:
```diff
@@ -0,0 +1,82 @@
+
+## Workflow
+
+One of the core objectives of the *Docs as Code* approach is to promote using the most suitable tools for each documentation task. As the saying goes, 'If your only tool is a hammer, every problem looks like a nail.' Understanding the purpose and selecting the right tools leads to more effective and valuable documentation.
```
Nitpick:

I would explain in more details the meaning and conclusion of the quote "If your only tool is a hammer, every problem looks like a nail" if you want to use it. 
Users might have enough information with the quote and the following sentence, but as they can come from any culture or background, I want to make sure the point is understandable for everyone.

NaraHita replied:
Sure. I can try and improve on this.

Sophie-Pages commented on website/events/community-hour-discussions/odh-015-docs-as-code.md:
```diff
@@ -0,0 +1,82 @@
+   
+- The core tools used are typically open-source and portable, rather than proprietary solutions (e.g., Adobe FrameMaker, MadCap, or RoboHelp). This avoids vendor lock-in, preventing dependency on a single vendor’s ecosystem that complicates switching or migration.
+  
+- Developers already know these tools- including IDEs, text editors, and scripting tools- making it more natural for them to contribute to documentation, fostering stronger collaboration.
```
Suggestion and question:
I suggest separating this block of bullet points into different paragraphs or topics. If you find that it's possible, subheaders can be useful to separate it into different topics. 

You've used quite a few lists and bullet points in the last part of this article (here and below as well). I would like you to ask yourself: 
- Are all the lists needed? 
- Is there a better way to show the information?

NaraHita replied:
Hi @Sophie-Pages , There was a lot of disjoint information given in the video. I thought putting it in list would help readers with retention of the main points. But I can change the sub list to paragraphs. What do you think works better?

Sophie-Pages replied:
That makes a lot of sense. Lists are a great way to skim through information easily. In that case, I find that the nested list about the **Positives** is a bit too long and actually difficult to skim through. What I would recommend is separating this nested list into different topics like automation, version control, maintenance, and tools (and add those as headers). 
The **Negatives** and **Practical tips** we can keep them as they are. 
What do you think?

NaraHita replied:
Okay. Will do that.

Sophie-Pages reviewed (changes requested):
Thank you @NaraHita for your work on this pull request. I’m very sorry for my late review. 

This is great work, and I like how you sometimes extended Robert’s ideas with your own. I’m interested to know why you’ve made the choice to frame it differently than Robert in some instances (I’ll add comments directly). 

For consistency, could you please follow the format that was used in the other articles? I’ll add a comment to give you an example.

Don't hesitate to ask questions on my suggestions or to disagree with me! I'll be happy to discuss documentation practices with you.

NaraHita committed:
Update website/events/community-hour-discussions/odh-015-docs-as-code.md

Co-authored-by: Sophie-Pages <34153782+Sophie-Pages@users.noreply.github.com>

NaraHita committed:
Update website/events/community-hour-discussions/odh-015-docs-as-code.md

Co-authored-by: Sophie-Pages <34153782+Sophie-Pages@users.noreply.github.com>

NaraHita committed:
Revise 'Docs as Code' document for clarity and focus

Updated the document to focus on getting hired as a technical writer, including presenter information and special thanks. Revised sections on the philosophy and workflow of 'Docs as Code' to clarify concepts and improve readability.

NaraHita committed:
Rename document title to 'Documentation as Code'

NaraHita commented:
Hi @Sophie-Pages , I have made some changes as per your comments. Please take a look and let me know on some of the points I have left comments on. 
Thanks again for all your feedback.

Sophie-Pages commented:
Thank you @NaraHita! I may not have the time to review your updates this week, but I'll try to do it before the end of next week! Thank you again for your patience.

NaraHita committed:
Merge branch 'canonical:main' into 269_Doc_as_Code

NaraHita committed:
Update odh-015-docs-as-code.md

NaraHita commented:
Hi @Sophie-Pages , I have updated the document as per your feedback. Thanks again for your help & support.
